### PR TITLE
fix: prevent double-pluralization and integrate compound words in H2 headings (#390, #383)

### LINKED ISSUE #383: H2 title generation issues: verb-first, compound word splitting, area prefix removal
## Problem Description

The H2 titles in tool family articles have three systematic issues:

1. **Doesn't start with a verb** — H2s should start with an action verb (Get, List, Create, Delete, Query, Run, Set, Update, Send, etc.)
2. **Doesn't turn compound CLI commands into multiple words** — Commands like 'loganalytics' should become 'Log Analytics', 'instrumentationkey' → 'instrumentation key', 'createorupdate' → 'Create or Update', etc.
3. **Still uses old Area prefix pattern** — Old pattern was like 'Workbooks: List workbooks' or 'Metrics: Query metrics'. Should be just the action without the area prefix.

## Example from PR #8647 (Azure Monitor)

| Current H2 | CLI Command | Problems | Suggested Fix |
|---|---|---|---|
| Get monitor activitylogs | monitor activitylog list | Issue #2 | List Activity Logs |
| Definitions monitor metrics | monitor metrics definitions | Issues #1, #3 | List Metric Definitions |
| Orchestrator next monitor instrumentation | monitor instrumentation orchestrator-next | Issues #1, #2 | Get Next Orchestrator Step |
| Createorupdate monitor webtests | monitor webtests createorupdate | Issues #1, #2 | Create or Update Web Tests |
| Workbooks: List workbooks | (workbook list) | Issue #3 | List Workbooks |
| Workbooks: Update workbook | (workbook update) | Issue #3 | Update Workbook |

## Impact

This affects **ALL 21 H2 titles in azure-monitor.md** (16 monitor tools + 5 workbook tools):
- 9 titles have Issue #1 (verb-first problem)
- 12 titles have Issue #2 (compound word splitting)
- 5 titles have Issue #3 (area prefix pattern)

Monitor is the largest tool family with the most tools, but this issue affects ALL tool family articles across all 52 Azure services.

## Where to Fix

The H2 titles are generated by the C# generators:
- **Primary location**: 'docs-generation/CSharpGenerator/' or equivalent generator projects
- **Template files**: '*.hbs' files that render H2 headings (likely 'tool-family-page.hbs' or similar)
- **Title generation logic**: The code that converts CLI command names (e.g., 'monitor activitylog list') into human-readable H2 titles

The fix needs to:
1. Extract the verb/action from the CLI command (first or last word depending on command structure)
2. Apply compound word splitting (e.g., 'activitylog' → 'Activity Log')
3. Remove the area prefix if present (e.g., 'Workbooks:' prefix)
4. Build the title as: '[Verb] [Area] [Action]' (e.g., 'List Activity Logs' not 'Get monitor activitylogs')

## Root Cause

The title generation likely:
- Takes the entire CLI command string and converts it directly without parsing
- Doesn't apply compound word transformations (using the 'compound-words.json' config)
- Still implements the old 'Area: Verb' pattern instead of verb-first pattern

### LINKED ISSUE #390: databaseses misspelling
When generating the postgresql tool family, the H2 was changed to included databaseses.  Investigate how the issue happened and what can be done to fix it in the future in the content generation pipeline 

Image

### PR DESCRIPTION
**Pull request by diberry**
## Summary

Fixes two related bugs in `DeterministicH2HeadingGenerator.cs` that affect H2 heading quality across all 52 namespaces.

### Bug #390 - databaseses misspelling
The `Pluralize()` method double-pluralized words already ending in "s" (e.g., "databases" became "databaseses").

**Fix:** Source-aware pluralization - resources extracted from CLI descriptions (already in natural form) skip pluralization. Added `IsLikelyPlural()` as a belt-and-suspenders safety net.

### Bug #383 - H2 title generation issues
Three systematic problems:
1. **Compound words not split** - `compound-words.json` was not integrated into heading generation
2. **Namespace prefix prepended** - single-word resources got namespace prepended
3. **Pluralization bug** - same root cause as #390

**Fixes:**
- Integrated `compound-words.json` via parameter passing (no mutable static state)
- Added `loganalytics` and `createorupdate` to `compound-words.json`
- Removed namespace prefix logic entirely - disambiguation algorithm handles collisions
- Both `BootstrapStep` and `CleanupGenerator` load and pass compound words

## Changes (5 files)
- `DeterministicH2HeadingGenerator.cs` - Core fixes
- `DeterministicH2HeadingGeneratorTests.cs` - 17 new TDD test cases
- `BootstrapStep.cs` - Pass compound words to generator
- `CleanupGenerator.cs` - Pass compound words to generator
- `compound-words.json` - Added 2 missing entries

## Testing
- 17 new tests covering all three bug fixes
- Full suite passes: 2,000+ tests, 0 failures, 0 regressions
- TDD workflow followed per AD-007

## Follow-up noted
- `DeterministicExamplePromptGenerator.cs:266-273` has a similar naive Pluralize - separate issue

Closes #390
Closes #383

### CODE REVIEW

**Review by diberry — COMMENTED** [review]
## Squad Review — APPROVE WITH SUGGESTIONS

No P0 or P1 findings. Three P2 suggestions and one nit below as inline comments.

**Strengths:**
- Source-aware pluralization is a clean design (avoids fragile English heuristics)
- Compound words passed as parameters (no mutable static state) 
- 17 new tests with good edge case coverage
- Full suite passes (2,000+ tests, 0 regressions)
- TDD workflow followed (AD-007 compliant)
- `FetchUpstreamFileAsync` has proper cancellation and local fallback

**Follow-up noted:** `DeterministicExamplePromptGenerator.cs:266-273` has same naive Pluralize — separate issue.

File: docs-generation/DocGeneration.Core.Shared/DeterministicH2HeadingGenerator.cs [hunk 1]
```diff
@@ -255,9 +252,13 @@ private static bool AreSimilar(string a, string b)
             || b.StartsWith(a, StringComparison.OrdinalIgnoreCase);
```
Comment (diberry) [inline docs-generation/DocGeneration.Core.Shared/DeterministicH2HeadingGenerator.cs]: nit: With the namespace prefix removal, `AreSimilar()` may now be dead code — its only caller was the removed block. If nothing else references it, consider cleaning it up.

**Review by diberry — COMMENTED** [review]
## Squad Team Review — APPROVE

**Reviewers:** Morgan (C#), Cameron (Tests), Riley (Architecture)

### Summary
Fixes double-pluralization ('databaseses') and integrates compound words into H2 heading generation. Source-aware pluralization distinguishes command-sourced vs description-sourced resources. 5 files changed, 17 new tests.

### Findings
- ✅ Correctness: Source-aware pluralization logic is sound
- ✅ Tests: 32 tests pass, comprehensive coverage
- ⚪ Nit: IsLikelyPlural has minor heuristic limitation for rare words ending in 'ns'/'ws' (e.g., 'lens', 'news') — acceptable trade-off for Azure CLI resource names

### Verdict: **Ready to merge** ✅
